Signal/src/ViewControllers/ConversationView/ConversationViewController+GestureRecognizers.swift

```diff
@@ -72,11 +65,18 @@ extension ConversationViewController: UIGestureRecognizerDelegate {
             return gestureRecognizer == collectionViewTapGestureRecognizer
         }
 
-        if gestureRecognizer == collectionViewPanGestureRecognizer {
+        if gestureRecognizer === collectionViewPanGestureRecognizer {
             // Only allow the pan gesture to recognize horizontal panning,
             // to avoid conflicts with the conversation view scroll view.
             let translation = collectionViewPanGestureRecognizer.translation(in: view)
-            return abs(translation.x) > abs(translation.y)
+            guard abs(translation.x) > abs(translation.y) else { return false }
+
+            // If there's nothing available to handle the pan, don't bother recognizing.
+            // This allows other recognizers like the navigationController popGesture to begin.
+            guard findPanHandler(sender: collectionViewPanGestureRecognizer) != nil else { return false }
```

Comment:
I don't love this, but it accomplishes what it needs to without changing a bunch of code. Basically, the pan should win if there's something to pan. Otherwise, it should defer elsewhere and allow another recognizer (e.g. interactive pop) to win.